PennMobile/Notifications/Rich_Notifications/GSRNotificationExtension.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  GSRNotificationExtension.swift
+//  PennMobile
+//
+//  Created by Dominic Holmes on 2/14/20.
+//  Copyright © 2020 PennLabs. All rights reserved.
+//
+
+import UIKit
+import UserNotifications
+
+class NotificationService: UNNotificationServiceExtension {
+    
+    // IMPORTANT: To take advantage of this extension, set the 'mutable-content' flag to '1'
+    /*
+     Use this to test: https://github.com/onmyway133/PushNotifications
+     Dominic's Device TokenId: EBA612F860AA80504F263BEBDBB1DC01DDE14C4B7D45E88ADA6BE300CD9D1BD2
+     EXAMPLE NOTIFICATION PAYLOAD:
+     {
+         "aps": {
+             "alert": {
+                 "body": "This is the body text, concerning your GSR in WIC RM 123 from 9:30-11am.",
+                 "title": "Upcoming GSR",
+                 "subtitle": "WIC Rm 129"
+             },
+             "mutable-content": 1,
+             "thread-id": 1,
+             "category": "UPCOMING_GSR"
+         },
+         "media-url": "https://snworksceo.imgix.net/dpn/20c5f2b4-9f36-4c47-985b-5b9673f86d77.sized-1000x1000.jpg?w=1000",
+         "sound": "default"
+     }
```

Comment:
From [this documentation](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CreatingtheNotificationPayload.html), it seems like custom data should be passed through a key containing `acme` in it. Is that correct? If so, I'm thinking I pass you the reservation info, which includes the booking ID, there.

Reply:
I think custom keys can just be specified at the top level. ‘acme’ can be any key.

I’m already using “media-url” as a custom key.

Reply:
Would be great to have the res id and necessary stuff to cancel the booking, and also the room name etc